crates/hyfetch/src/pride_month.rs

```diff
@@ -0,0 +1,228 @@
+//by 1337isnot1337
+
+use crate::{
+    color_util::{color, ForegroundBackground, ToAnsiString},
+    presets::Preset,
+    types,
+};
+use palette::{encoding::Srgb, rgb::Rgb};
+use std::{
+    io::{self, Write},
+    process,
+    sync::atomic::{AtomicBool, Ordering},
+    thread,
+    time::Duration,
+};
+use types::AnsiMode;
+
+use term_size::dimensions;
+
+static KEY_PRESSED: AtomicBool = AtomicBool::new(false);
+
+#[allow(clippy::too_many_lines)]
+pub fn start_animation() {
+    let mut input: String = String::new();
+    //this thread listens for input
+    let _listen_thread = thread::spawn(move || {
+        match io::stdin().read_line(&mut input) {
+            Ok(0) => {},
+            Ok(_) => KEY_PRESSED.store(true, Ordering::SeqCst),  // Input detected
+            Err(e) => unreachable!("I hope you can't reach this. If so there was an error with reading from stdin: {e}"),  // Error reading input
+        }
+    });
+
+    // the "happy pride month" text that is displayed
+    let text = r".======================================================.
```

Comment:
Make this a `const` at the file / mod level. Include newlines at beginning and end of the string:

```rust
const TEXT_ASCII: &str = r############################################################"
.======================================================.
| .  .              .__       .     .  .       , .   | |
| |__| _.._ ._   .  [__)._.* _| _   |\/| _ ._ -+-|_  | |
| |  |(_][_)[_)\_|  |   [  |(_](/,  |  |(_)[ ) | [ ) * |
|        |  |  ._|                                     |
'======================================================'
"############################################################;
```

And just do something like this when you use it:

```rust
let text = &TEXT_ASCII[1..TEXT_ASCII.len() - 1];
```